index.html

```diff
@@ -469,18 +487,194 @@ <h4>Attributes</h4>
               </p>
             </li>
             <li>
-              <p>If <var>track</var>.{{MediaStreamTrack/[[VideoStats]]}} is
+              <p>If <var>track</var>.{{MediaStreamTrack/[[Stats]]}} is
               <code>null</code>, throw {{NotSupportedError}} and abort these
               steps.</p>
             </li>
             <li>
-              <p>Return
-              <var>track</var>.{{MediaStreamTrack/[[VideoStats]]}}.</p>
+              <p>Return <var>track</var>.{{MediaStreamTrack/[[Stats]]}}.</p>
             </li>
           </ol>
         </dd>
       </dl>
     </section>
+  </section>
+    <section>
+      <h4>The MediaStreamTrackAudioStats interface</h4>
+      <div>
+        <pre class="idl" data-cite="HR-TIME">[Exposed=Window]
+interface MediaStreamTrackAudioStats {
+  readonly attribute unsigned long long deliveredFrames;
+  readonly attribute DOMHighResTimeStamp deliveredFramesDuration;
+  readonly attribute DOMHighResTimeStamp deliveredFramesDelay;
+  readonly attribute unsigned long long droppedFrames;
+  readonly attribute DOMHighResTimeStamp droppedFramesDuration;
+  [Default] object toJSON();
+};
+        </pre>
+        <div>
+          <p>The {{MediaStreamTrackAudioStats}} expose frame counters for the
+          {{MediaStreamTrack}} that created it. For this track, the user agent
+          is required to count each audio frame from its source as follows:</p>
+          <ul>
+            <li>
+              <p>A frame is considered a <dfn data-lt="delivered audio frames">
+              delivered audio frame</dfn> if it either was delivered to a sink
+              or would have been delivered to a sink, if one was connected.</p>
+            </li>
+            <li>
+              <p>The <dfn data-lt="delivered audio frames duration">delivered
+              audio frames duration</dfn> is the total duration of all
+              [= delivered audio frames =], in milliseconds.</p>
+            </li>
+            <li>
+              <p>The <dfn data-lt="delivered audio frames delay">delivered audio
+              frames delay</dfn> is the sum of [= delivered audio frames =]'s
+              capture delays. The capture delay is the estimated time between
+              the capture device emitting the frame and the frame being
+              considered delivered, in milliseconds.</p>
+            </li>
+            <div class="note">
+              <p>The average capture delay per audio frame can be calculated as
+              [= delivered audio frames delay =] / [= delivered audio
+              frames =].</p>
```

Comment:
> Average, like most aggregation methods, are masking large but transient changes.

Only if you calculate the average of the entire session, which I think some apps would still be interested in. But you can calculate the average over any period of time that you're interested, such as average over the last second.

Just poll `track.stats` at two points in time and take the `delta delivered delay / delta delivered frames`. If you do this every second you'll have a very good understanding of how the delay changes over time. We have tools for building graphs over time, and our app (Google Meet) is interested in both 1 Hz polling, 0.1 Hz polling and 10 Hz polling, depending on use cases.

> Latency also is not a characteristic of an audio frame

If latency changes over time, you'll have different latencies that is active when different audio frames are being passed through the pipeline, leading to different per-frame latencies. It's important that the aggregate latency counter is incremented at the same time as the aggregate delivered frames counter, perhaps I should clarify that in the PR.

Reply:
> Only if you calculate the average of the entire session, which I think some apps would still be interested in. But you can calculate the average over any period of time that you're interested, such as average over the last second.

This needs to be specced if it's not a normal average but a rolling average with a particular width.

> Just poll track.stats at two points in time and take the delta delivered delay / delta delivered frames. If you do this every second you'll have a very good understanding of how the delay changes over time. We have tools for building graphs over time, and our app (Google Meet) is interested in both 1 Hz polling, 0.1 Hz polling and 10 Hz polling, depending on use cases.

How can you do this if you don't know if you have an average, rolling average, or something else? Besides, it's still nonsensical. If I have three audio buffers delivered, of 1024 frames each, and latencies are respectively 10 frames, 500 frames and 36 frames, it goes:

(10 + 500 + 36) / (1024 + 1024 + 1024) = 0.177734375

this number doesn't mean anything. (10 + 500 + 36) / 3 means something, that's a regular average, but it masks jitter by definition because it's an average, so it's a low pass.

> If latency changes over time, you'll have different latencies that is active when different audio frames are being passed through the pipeline, leading to different per-frame latencies. It's important that the aggregate latency counter is incremented at the same time as the aggregate delivered frames counter, perhaps I should clarify that in the PR.

It seems that you're using a non-standard definition of an audio frame. An [audio frame](https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Audio_concepts#audio_channels_and_frames) is the group of audio samples that happen at the same time, one per audio channel.

You maybe way to say audio buffer or something like that. But still this isn't defined properly and not what is needed.

Reply:
I'm not sure I follow. If I make 10 measurements I can get the average measurement as total / 10. In this case, the number of measurements is deliveredFrames and each individual measurement is the latency of that delivered frame. So total latency shouldn't be any different than individual latencies, other than it contains multiple of them.

Is there something problematic about measuring latency of an individual audio frame?